dispatch/dispatch.go

```diff
@@ -133,7 +138,17 @@ func (d *Dispatcher) Groups(matchers []*labels.Matcher) AlertOverview {
 				if !a.EndsAt.IsZero() && a.EndsAt.Before(now) {
 					continue
 				}
+
 				status := d.marker.Status(a.Fingerprint())
+
+				if !showSilenced && len(status.SilencedBy) != 0 {
+					continue
+				}
+
+				a.Annotations = a.Annotations.Merge(
+					model.LabelSet{"receiver": model.LabelValue(route.RouteOpts.Receiver)},
```

Comment:
Hm, would that be an alert from a different aggrGroup, which would have a different route (and route receiver)? I'm reading the code and trying to make sense of it -- if an aggrGroup is defined by its set of labels, then that would mean the alert is at a different point in the tree, and potentially included?

In this case it would be a duplicate (except for the different `receiver` field), so yeah a list would be right. I totally forgot about `continue`. So we'll want to de-duplicate outside this method inside the poorly named `flattenGroups` function in `api.go`, I think.

>maybe go the other way and have an index of the receivers to the leaf routes/aggregation groups

It seems like from line 120 we already have that:

```go
for route, ags := range d.aggrGroups {
```

>This makes selecting multiple receivers more difficult though.

I think this is making an assumption about the UX, that we're going to have a multi-select html element. But even if we go this route, I think we could just squish all the selected receivers into a regexp `(receiverA|receiverB|receiverF)` when making the API call, and the current regexp match method would work.